packages/material-ui/src/Modal/Modal.js

```diff
@@ -140,8 +140,13 @@ class Modal extends React.Component {
     this.modalRef.scrollTop = 0;
   };
 
-  handleClose = () => {
-    this.props.manager.remove(this);
+  handleClose = reason => {
+    const hasTransition = getHasTransition(this.props);
+    /* If the component has a transition and is not unmounting let the transition handle
+    removing the style this prevents elements moving around when the Modal is closed. */
+    if (!hasTransition || reason === 'unmount') {
+      this.props.manager.remove(this);
```

Comment:
Can this become an issue since the manager can still access `this` after the transition has completed?

Reply:
Hmm, I can see the need for concern. I'm not sure though I'd also be interested in the answer to this. The Modal should still have a valid `this` after the transition has completed though since the Modal itself wouldn't unmount and if it did it would call componentDidUnmount and attempt to remove it straight away. If one of them isn't found in the list modal manager will handle it.

Reply:
The manager does not assume that all the modals it holds are open? If so then everything should be fine since we always remove on unmount.